node.gyp

```diff
@@ -720,6 +720,38 @@
   ], # end targets
 
   'conditions': [
+    [ 'node_target_type=="static_library"', {
+      'targets': [
+        {
+          'target_name': 'static_node',
+          'type': 'executable',
+          'product_name': 'node',
+          'dependencies': [
+            '<(node_core_target_name)',
+          ],
+          'sources+': [
+            'src/node_main.cc',
+          ],
+          'include_dirs': [
+            'deps/v8/include',
+          ],
+          'xcode_settings': {
+            'OTHER_LDFLAGS': [
+	      '-Wl,-force_load,<(PRODUCT_DIR)/libnode.a',
+            ],
+          },
+          'conditions': [
+            ['OS in "linux freebsd"', {
```

Comment:
You're right, this does look like it's missing a couple of platforms where `--whole-archive` can be used. We also need to disable the static `node` build on platforms that don't support.